cmd/prepare-storage.go

```diff
@@ -135,19 +136,12 @@ func prepForInitXL(firstDisk bool, sErrs []error, diskCount int) InitActions {
 	disksOffline := errMap[errDiskNotFound]
 	disksFormatted := errMap[nil]
 	disksUnformatted := errMap[errUnformattedDisk]
```

Comment:
@harshavardhana, this PR may fix the issue but `prepForInitXL` is difficult to read and has been source of bugs like the one you are fixing. So, to make sure we don't leave behind bugs like these I have tried to list all user-visible states and corresponding error conditions. If this exercise comes in the way of fixing a known bug in search of bugs that are yet to be discovered by testing or user, we can open a new issue to track the readability problem and take this PR as is. Let me know what you think.

## Possible states and conditions

### Formatted setup
- InitObjectLayer
`disksFormatted >= readQuorum`

- Wait for quorum
`disksFormatted < readQuorum && disksFormatted + disksOffline >= readQuorum`
(we don't know yet if there are unformatted disks)
  - Wait for heal
` disksFormatted >= readQuorum && disksUnformatted > 0`
(here we know there is at least one unformatted disk which requires healing)

### Unformatted setup
- Format/Wait for format
`disksUnformatted == diskCount`
- Wait for all
`disksUnformatted + disksOffline == disksCount`

All other conditions should lead to abort of server.